hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageDispatcherService.java

```diff
@@ -500,4 +505,58 @@ private List<SoftwareModuleMetadata> getSoftwareModuleMetadata(final SoftwareMod
                 PageRequest.of(0, RepositoryConstants.MAX_META_DATA_COUNT), module.getId()).getContent();
     }
 
+    private void sendUpdateMessageToTargets(final Map<String, ActionProperties> actions, final List<Target> targets,
+                                            final Map<SoftwareModule, List<SoftwareModuleMetadata>> modules) {
+
+            List<DmfTarget> dmfTargets = targets.stream().filter(target -> IpUtil.isAmqpUri(target.getAddress()))
+                    .map(t -> convertToDmfTarget(t, actions.get(t.getControllerId()).getId()))
+                    .collect(Collectors.toList());
+
+            final DmfBatchDownloadAndUpdateRequest batchRequest = new DmfBatchDownloadAndUpdateRequest();
+            batchRequest.setTimestamp(System.currentTimeMillis());
+            batchRequest.setTargets(dmfTargets);
+
+            if (modules != null) {
+                modules.entrySet().forEach(entry ->
+                        batchRequest.addSoftwareModule(convertToAmqpSoftwareModule(targets.get(0), entry)));
+            }
+
+        Optional<ActionProperties> first = actions.values().stream().findFirst();
+
+
+        final Message message = getMessageConverter().toMessage(batchRequest,
+                    createMessagePropertiesBatch(first.get().getTenant(), getBatchEventTopicForAction(first.get())));
+            amqpSenderService.sendMessage(message, targets.get(0).getAddress());
+
+    }
+
+    protected DmfTarget convertToDmfTarget(Target target, Long actionId) {
+        DmfTarget dmfTarget = new DmfTarget();
+        dmfTarget.setActionId(actionId);
+        dmfTarget.setControllerId(target.getControllerId());
+        dmfTarget.setTargetSecurityToken(systemSecurityContext.runAsSystem(target::getSecurityToken));
+        return dmfTarget;
+    }
+
+    private static MessageProperties createMessagePropertiesBatch(final String tenant, final EventTopic topic) {
+        final MessageProperties messageProperties = new MessageProperties();
+        messageProperties.setContentType(MessageProperties.CONTENT_TYPE_JSON);
+        messageProperties.setHeader(MessageHeaderKey.CONTENT_TYPE, MessageProperties.CONTENT_TYPE_JSON);
+        messageProperties.setHeader(MessageHeaderKey.TENANT, tenant);
+
+        messageProperties.setHeader(MessageHeaderKey.TOPIC, topic);
+        messageProperties.setHeader(MessageHeaderKey.TYPE, MessageType.EVENT);
+        return messageProperties;
+    }
+
+    public boolean isBatchAssignmentsEnabled() {
+        return systemSecurityContext.runAsSystem(() -> tenantConfigurationManagement
+                .getConfigurationValue(BATCH_ASSIGNMENTS_ENABLED, Boolean.class).getValue());
+    }
+
+    private static EventTopic getBatchEventTopicForAction(final ActionProperties action) {
+        return (Action.ActionType.DOWNLOAD_ONLY == action.getActionType() || !action.isMaintenanceWindowAvailable())
```

Comment:
Is this logic correct? So - if there is no maintenance window specified for the action, it would always be Download-Only, even if ActionType would be Download-And-Install?

Reply:
I copied the logic from both Single and Multi assignments executions.
"* Method to get the type of event depending on whether the action is a
     * DOWNLOAD_ONLY action or if it has a valid maintenance window available or
     * not based on defined maintenance schedule. In case of no maintenance
     * schedule or if there is a valid window available, the topic "
 I assume it should be consistent with them. Tell me if I need to change it